src/routes/trainingSession/trainingSession.controller.ts

```diff
@@ -0,0 +1,71 @@
+import { TrainingSession } from "../../schemas/training.session";
+import { Request, Response } from 'express';
+import { TrainingSessionDocument, TTrainingSession } from "../../types/db/training.session.types";
+
+export async function getTrainingSession(
+    req: Request,
+    res: Response
+    ){
+        try {
+            let trainingSessionDocs: (TrainingSessionDocument & Required<{ _id: TTrainingSession; }>)[] = [];
+
+            if (req.query.userId == null) {
+                throw new Error("No userId provided!")
+            }
+
+            if ( Array.isArray(req.query.id) ) {
+                req.query.id.forEach(async id => {
+                    const singleSession = await TrainingSession.findById(id)
+                    .populate({
+                        path: 'executions',
+                        populate: {
+                            path: 'exercise',
+                            model: 'Exercise'
+                        }
+                    })
+                    if (singleSession) {
+                        trainingSessionDocs.push(singleSession);
+                        
+                    }
+                })
+                    
+            } else {
+                trainingSessionDocs = await TrainingSession.find({
+                    userId: req.query.userId
+                })
+                .populate({
+                    path: 'executions',
+                    populate: {
+                        path: 'exercise',
+                        model: 'Exercise'
+                    }
+                })
+            }
```

Comment:
Wenn das id Feld der Request ein Array ist, dann wird nach den korrespondierenden Trainingssessions gesucht. Wenn das id Feld der Request kein Array ist, dann wird nach allen Trainingssessions für eine userId gesucht. 

Falls du entweder mehrere bestimmte Trainingssessions oder alle für einen user bekommen willst wäre das verständlich, dann solltest du aber bei der oberen Query noch hinzufügen, dass ein User nur seine eigenen Trainingssessions querien kann.  (Habs jetzt hier geschrieben und markiert, weil ich nicht zwei Kommentare übereinander wollte :D)

Ich persönlich spalte sowas immer in zwei routes auf, eine getSingle und eine getAll (jeweils mit User Protection) und wenn das Frontend 3 bestimmte Trainingssessions braucht, dann schicken sie halt 3 Anfragen an die getSingle Route. Ist aber Geschmackssache denke ich